platform/linuxbsd/wayland/wayland_thread.cpp

```diff
@@ -4400,6 +4432,44 @@ void WaylandThread::pointer_set_hint(const Point2i &p_hint) {
 	}
 }
 
+void WaylandThread::pointer_warp(const Point2i &p_to) {
+	// NOTE: This is for compositors that don't support the pointer-warp protocol.
+	// It's hacked together and not guaranteed to work.
+	if (registry.wp_pointer_warp == nullptr) {
+		PointerConstraint old_constraint = pointer_get_constraint();
+
+		pointer_set_constraint(PointerConstraint::LOCKED);
+		pointer_set_hint(p_to);
+
+		pointer_set_constraint(old_constraint);
+
+		return;
+	}
+
+	SeatState *ss = wl_seat_get_seat_state(wl_seat_current);
+	if (!ss) {
+		return;
+	}
+
+	WindowState *ws = window_get_state(ss->pointer_data.pointed_id);
+
+	int wl_pos_x = 0;
+	int wl_pos_y = 0;
+
+	if (ws) {
+		// NOTE: It looks like it's not really recommended to convert from
+		// "godot-space" to "wayland-space" and in general I received mixed feelings
+		// discussing about this. I'm not really sure about the maths behind this but,
+		// oh well. ¯\_(ツ)_/¯
+		// See: https://oftc.irclog.whitequark.org/wayland/2023-08-23#1692756914-1692816818
+		wl_pos_x = std::round(p_to.x / window_state_get_scale_factor(ws));
+		wl_pos_y = std::round(p_to.y / window_state_get_scale_factor(ws));
+	}
```

Comment:
Probably should be:

```c++
	if (!ws) {
		return;
	}

	int wl_pos_x = std::round(p_to.x / window_....
```

Warping to 0,0 is not expected behavior if there's no active window.

Reply:
it's been a while since I wrote the original code but yea, I guess that makes more sense.